include/clang/AST/Stmt.h

```diff
@@ -252,18 +252,11 @@ class LLVM_ALIGNAS(LLVM_PTR_SIZE) Stmt {
     unsigned NumArgs : 32 - 8 - 1 - NumExprBits;
   };
 
-  class CountBoundsExprBitFields {
-    friend class CountBoundsExpr;
+  class BoundsExprBitFields {
+    friend class BoundsExpr;
 
     unsigned : NumExprBits;
-    unsigned Kind : 1;
-  };
-
-  class NullaryBoundsExprBitFields {
-    friend class NullaryBoundsExpr;
-
-    unsigned : NumExprBits;
-    unsigned Kind : 1;
+    unsigned Kind : 3;
```

Comment:
Are there any safeguards in place that would catch someone adding to the Kind type but not updating the length of the bitfield?  IMHO, compile-time asserts are great for this sort of thing--possibly adding a sentinel value to the enum type, using a #define for the bit-field length, and a static assert to compare them with appropriate bit arithmetic.